libraries/botbuilder-ai/src/luisRecognizer.ts

```diff
@@ -751,4 +751,16 @@ export class LuisRecognizer implements LuisRecognizerTelemetryClient {
             throw new Error(`Invalid \`endpointKey\` value detected: ${this.application.endpointKey}\nPlease make sure your endpointKey is a valid LUIS Endpoint Key, e.g. "048ec46dc58e495482b0c447cfdbd291".`);
         }
     }
+
+    /**
+     * Allows users to pass in either Machine Name (LUIS region or custom <name>.api...) or the full URL
+     * 
+     * @param endpoint https://<region/custom>.api.cognitive.microsoft.com
+     */
+    private getValidEndpoint(endpoint: string): string|null {
```

Comment:
I can envision an endless series of strings that pass the Regex, but don't work with the pattern. 

My opinion is we should take the full URI endpoint from the dev and just use it "as-is". The parsing game here is a losing battle, and will only cause issues...

Reply:
My understanding of @diberry 's email regarding the new endpoint is that it changes from `<region>.api...` to `<custom>.api`, which this still covers.

That being said, if we at least sync up the samples and make it clear what it's asking for, I'm happy. To poorly summarize a discussion with @Kaiqb today, "I've stopped writing LUIS bots because I always forget how to pass the endpoint". We see maybe an issue every week or so regarding this, too.